No more can define `__filename`, `__dirname` variables

### Describe the bug

I can no more can define `__filename`, `__dirname` variables in vite.config.js in a "module" project.

It worked with Vite 2.5.7. 
For example: [this](https://github.com/AlttiRi/directory-snapshot-explorer/blob/2bfedf1e27d4d91f06b5a75eafcadd11f996fb48/vite.config.js#L4-L8) ([Vite 2.5.7](https://github.com/AlttiRi/directory-snapshot-explorer/blob/2bfedf1e27d4d91f06b5a75eafcadd11f996fb48/package.json#L23)) is successfully built.

Now I get the error:
```
 > vite.config.js:7:6: error: Expected identifier but found "\"C:\\\\Dev\\\\vite-server-json-gz\\\\vite.config.js\""
    7 │ const "C:\\Dev\\vite-server-json-gz\\vite.config.js" = fileURLToPath("file://C:\\Dev\\vite-server-json-gz\\vite.config.js");
      ╵       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

failed to load config from C:\Dev\vite-server-json-gz\vite.config.js
error during build:
Error: Build failed with 1 error:
vite.config.js:7:6: error: Expected identifier but found "\"C:\\\\Dev\\\\vite-server-json-gz\\\\vite.config.js\""
...
```

---

For the reproduction uncomment [these lines](https://github.com/AlttiRi/serve-json-gz-length-test/blob/f21d9528a7ba79ba684b261a862d420b1fc64c58/vite.config.js#L4-L8) of [the minimal reproduction repo](https://github.com/AlttiRi/serve-json-gz-length-test).


### Reproduction

https://github.com/AlttiRi/serve-json-gz-length-test

### System Info

```shell
OS: Windows 10 10.0.19043
Node: 17.3.0
Yarn: 1.22.5
npm: 7.19.1
vite: 2.7.2
```


### Used Package Manager

npm

### Logs

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitejs/vite/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitejs/vite/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitejs.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitejs/vite/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a Vite issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/vue-next instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitejs/vite/discussions) or join our [Discord Chat Server](https://chat.vitejs.dev/).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/docs/config/index.md
+++ b/docs/config/index.md
@@ -30,9 +30,8 @@ vite --config my-config.js
 Note Vite will replace `__filename`, `__dirname`, and `import.meta.url`. Using these as variable names will result in an error:
 
 ```js
-// example
 const __filename = "value"
-// will compile with
+// will be transformed to
 const "path/vite.config.js" = "value"
 ```
 

--- a/docs/config/index.md
+++ b/docs/config/index.md
@@ -27,7 +27,7 @@ You can also explicitly specify a config file to use with the `--config` CLI opt
 vite --config my-config.js
 ```
 
-Note Vite will replace `__filename`, `__dirname`, and `import.meta.url`. Using these as variable names will result in an error:
+Note that Vite will replace `__filename`, `__dirname`, and `import.meta.url`. Using these as variable names will result in an error:
 
 ```js
 const __filename = "value"

--- a/docs/config/index.md
+++ b/docs/config/index.md
@@ -27,9 +27,7 @@ You can also explicitly specify a config file to use with the `--config` CLI opt
 vite --config my-config.js
 ```
 
-Note Vite will replace `__filename`/`__dirname`/`import.meta.url` with the correct values.
-
-Using `__filename`/`__dirname` as variable names will be an error:
+Note Vite will replace `__filename`, `__dirname`, and `import.meta.url`. Using these as variable names will result in an error:
 
 ```js
 // example

--- a/docs/config/index.md
+++ b/docs/config/index.md
@@ -27,6 +27,17 @@ You can also explicitly specify a config file to use with the `--config` CLI opt
 vite --config my-config.js
 ```
 
+Note Vite will replace `__filename`/`__dirname`/`import.meta.url` with the correct values.
+
+Using `__filename`/`__dirname` as variable names will be an error:
+
+```js
+// example
+const __filename = "value"
+// will compile with
+const "path/vite.config.js" = "value"
+```
+
 ### Config Intellisense
 
 Since Vite ships with TypeScript typings, you can leverage your IDE's intellisense with jsdoc type hints:

